[Reviewed] markdown preview

kengo-mashita opened the pull request:
## チケット

https://report.r-jc.jp/issues/24445

## 内容

- マークダウン表示用のライブラリを追加
- 最低限パースして表示するようにした

## できていないこと

- アプリバー（アクションバー）周り
- スクロールビュー
- 編集画面との連携

## ライブラリについて

[Markwon](https://github.com/noties/Markwon)

公式のサンプルがあるので、実装をさらに進めることは可能
[サンプルアプリ](https://github.com/noties/Markwon/tree/master/app-sample)

kengo-mashita committed:
[WIP] markdown preview

diesel141 commented on .idea/gradle.xml:
```diff
@@ -7,7 +7,6 @@
         <option name="testRunner" value="PLATFORM" />
         <option name="distributionType" value="DEFAULT_WRAPPED" />
         <option name="externalProjectPath" value="$PROJECT_DIR$" />
-        <option name="gradleJvm" value="12" />
```
[must] idea/配下のファイルはpushしないでください。.gitignoreで制御しているはずなので、
developが最新のものであるか確認願います。

diesel141 commented on app/src/main/res/layout/activity_preview.xml:
```diff
@@ -0,0 +1,15 @@
+    tools:context=".Preview">
+
+    <TextView
+        android:id="@+id/textView"
```
[must] idは示すものが明確にわかるものに命名してください

diesel141 commented on app/src/main/res/layout/activity_preview.xml:
```diff
@@ -0,0 +1,15 @@
+        android:layout_height="88dp"
+        android:text="TextView" />
+
+</androidx.coordinatorlayout.widget.CoordinatorLayout>
```
[must] xmlの末尾は改行を挿入してください。

diesel141 commented on app/src/main/java/jp/co/lightmarkdown_android/Preview.kt:
```diff
@@ -0,0 +1,30 @@
+import io.noties.markwon.Markwon
+import java.security.AccessController.getContext
+
+class Preview : AppCompatActivity() {
```
[Must] Activityの名称は「PreviewActivity」にしてください。機能なのかFragmentなのか画面なのか明確化する必要があります。

diesel141 commented on app/src/main/java/jp/co/lightmarkdown_android/Preview.kt:
```diff
@@ -0,0 +1,30 @@
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Button
```
[must] 不要なインポートは削除してください。optimize importすると良いです。

diesel141 commented on app/src/main/java/jp/co/lightmarkdown_android/Preview.kt:
```diff
@@ -0,0 +1,30 @@
+import android.widget.Button
+import android.widget.TextView
+import io.noties.markwon.Markwon
+import java.security.AccessController.getContext
```
[must] 不要なインポートは削除してください。optimize importすると良いです。

diesel141 commented on app/src/main/java/jp/co/lightmarkdown_android/Preview.kt:
```diff
@@ -0,0 +1,30 @@
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_preview)
+        val textView = findViewById<TextView>(R.id.textView)
```
[must] build.gradle の plugins に「id 'kotlin-android-extensions'」を追記してください。
Kotlinでは基本的に「findViewById」を使用しなくても、Activity内で直接idを参照できます。

diesel141 commented on app/src/main/res/layout/activity_preview.xml:
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.coordinatorlayout.widget.CoordinatorLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
[must] CoordinatorLayoutのライブラリを使用しないのであれば、シンプルに LinearLayout 等の最小限のViewGroupeを使ってください。

diesel141 reviewed (commented):
ご確認願います。

kengo-mashita committed:
レビュー指摘対応
・gradle.xmlを前の状態に戻す
・kotlin-android-extensionsの追加
・PreviewActivityにクラス名を変更
・不要なインポートを除去
・LinearLayoutに変更

kengo-mashita commented:
レビューありがとうございます。
下記コミットにて対応しました。
0fba23a573dc727dee4cb21c31dae5c8ce4affe2

diesel141 commented on app/src/main/java/jp/co/lightmarkdown_android/PreviewActivity.kt:
```diff
@@ -0,0 +1,27 @@
+      **bold _italic_ bold**
+    """.trimIndent()
+
+        markwon.setMarkdown(markdownPreviewContent, md);
```
[imo] 末尾のセミコロンは不要です。適当なタイミングで修正してください。

diesel141 reviewed (approved):
確認しました。[imo]1件だけですのでマージして良いです。
